File: Scripts/perl/mcp/mcp.pl
#!/usr/bin/env perl
use warnings;
use strict;

my $move;
my $searchStr;
my $replaceStr;

# display usage
sub usage
{
	print "\nMCP is a perl script that can be used to copy/move one\n";
	print "or more files, replacing a search string in the filename.\n";
	print "USAGE:\n";
	print "mcp [-m] SEARCH REPLACE FILES\n";
	print "\tWhere SEARCH is a string you are searching for,\n";
	print "\tREPLACE is a string you would like to replace\n";
	print "\tSEARCH with, and FILES is a list of files to search\n";
	print "\tthrough.\n";
	print "\tThe -m option specified that the files be moved instead of\n";
	print "\tcopied.\n\n";
}

# copies/moves a single file, if necessary
sub dupFile( $ )
{
	my $file = $_[0];
	
	if( $file =~ m/$searchStr/ )
	{
		my $newFile = "$file";
		$newFile =~ s/$searchStr/$replaceStr/;
		
		if( $move )
		{
			print "mv $file $newFile\n";
			system( "mv $file $newFile" );
		} else
		{
			print "cp $file $newFile\n";
			system( "cp $file $newFile" );
		}
	}
}

# make sure we have at least 3 arguments
if( scalar( @ARGV ) < 3 )
{
	usage();
} else
{
	if( $ARGV[0] eq "-m" )
	{
		$move = 1;
		shift( @ARGV );
	}
	
	$searchStr = shift( @ARGV );
	@replaceStr = shift( @ARGV );
	
	while( scalar( @ARGV ) )
	{
		dupFile( shift( @ARGV ) );
	}
}

File: Scripts/python/mcp/mcp.py
#!/usr/bin/python

import sys

from os import rename
from shutil import copyfile

def usage():
	"""
	Prints usage info.
	"""
	print '\nMCP is a python script that can be used to copy/move one'
	print 'or more files, replacing a search string in the filename.'
	print 'USAGE:'
	print 'mcp [-m] SEARCH REPLACE FILES'
	print '\tWhere SEARCH is a string you are searching for,'
	print '\tREPLACE is a string you would like to replace'
	print '\tSEARCH with, and FILES is a list of files to search'
	print '\tthrough.'
	print '\tThe -m option specified that the files be moved instead of'
	print '\tcopied.\n'

def dupFile( move, search, replace, filename ):
	"""
	Copies or moves a file, if necessary.
	"""
	if( filename.find( search ) != -1:
		newFilename = filename.replace( search, replace )
		
		if move:
			print 'mv', filename, '->', newFilename
			rename( filename, newFilename )
		else:
			print 'cp', filename, '->', newFilename
			copyfile( filename, newFilename )
	
	return False

def main():
	"""
	Entry point
	"""
	# Pop the first arg, which is the script name
	sys.argv.pop( 0 )
	
	# Make sure we have at least three args
	if( len( sys.argv ) < 3:
		usage()
	else
		move = False
		
		if sys.argv[0] == '-m':
			move = True
			sys.argv.pop( 0 )
		
		searchStr = sys.argv.pop( 0 )
		replaceStr = sys.argv.pop( 0 )
		
		for s in sys.argv[:]:
			dupFile( move, searchStr, replaceStr, s )
				
	sys.exit( 0 )

# Standard template to call the main function
if __name__ == '__main__'
	main()
